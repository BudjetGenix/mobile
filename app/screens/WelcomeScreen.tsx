/* eslint-disable global-require */
import React, { FC } from 'react';
import { View, StyleSheet, ImageBackground, Image } from 'react-native';

import Text from '@components/Text';
import Button from '@components/Button';
import colors from '@config/colors';
import { useAuth } from '@auth/context';

interface WelcomeScreenProps {}

const WelcomeScreen: FC<WelcomeScreenProps> = () => {
  const { login } = useAuth();
  return (
    <ImageBackground
      blurRadius={5}
      style={styles.background}
      source={require('@assets/background.jpg')}
    >
      <View style={styles.logoContainer}>
        <Image source={require('@assets/icon.png')} style={styles.logo} />
        <Text style={styles.tagLine}>Tag Line</Text>
      </View>
      <View style={styles.buttonsContainer}>
        <Button title="Login" color={colors.primary} onPress={login} />
        <Button
          title="Register"
          color={colors.secondary}
          onPress={() => console.log('Register')}
        />
      </View>
    </ImageBackground>
  );
};

const styles = StyleSheet.create({
  background: {
    flex: 1,
    justifyContent: 'flex-end',
    alignItems: 'center',
  },
  buttonsContainer: {
    padding: 20,
    width: '100%',
  },
  logo: {
    width: 100,
    height: 100,
  },
  logoContainer: {
    position: 'absolute',
    top: 75,
    alignItems: 'center',
  },
  tagLine: {
    fontSize: 20,
    fontWeight: '600',
    paddingVertical: 20,
  },
});

export default WelcomeScreen;
