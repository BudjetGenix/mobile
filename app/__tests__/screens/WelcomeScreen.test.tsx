import React from 'react';
import renderer from 'react-test-renderer';

import WelcomeScreen from '@screens/WelcomeScreen';

describe('<WelcomeScreen />', () => {
  it('renders properly', () => {
    const tree: any = renderer.create(<WelcomeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
