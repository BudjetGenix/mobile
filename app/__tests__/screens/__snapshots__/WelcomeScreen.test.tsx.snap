// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomeScreen /> renders properly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "flex-end",
    }
  }
>
  <Image
    blurRadius={5}
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "position": "absolute",
        "top": 75,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 100,
          "width": 100,
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#0c0c0c",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Object {
            "fontSize": 20,
            "fontWeight": "600",
            "paddingVertical": 20,
          },
        ]
      }
    >
      Tag Line
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": 20,
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fc5c65",
          "borderRadius": 25,
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "padding": 15,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 18,
            "fontWeight": "bold",
            "textTransform": "uppercase",
          }
        }
      >
        Login
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4ecdc4",
          "borderRadius": 25,
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "padding": 15,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 18,
            "fontWeight": "bold",
            "textTransform": "uppercase",
          }
        }
      >
        Register
      </Text>
    </View>
  </View>
</View>
`;
